platform/mbed_poll.h

```diff
@@ -0,0 +1,59 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_POLL_H
+#define MBED_POLL_H
+
+#define POLLIN         0x0001
+#define POLLOUT        0x0010
+
+#define POLLERR        0x1000
+#define POLLHUP        0x2000
+#define POLLNVAL       0x4000
```

Comment:
Minor nit: It would be worth documenting these defines, you can just grab the docs from the man page:
http://man7.org/linux/man-pages/man2/poll.2.html